### ISSUE TITLE
reference in metadata is documented as OPTIONAL, but build fails if it is absent

### ISSUE THREAD

**Issue by jasmdk** [getporter/porter#1354]
## Describe the bug
The 'reference' field is documented as OPTIONAL and will take precedence if present, but if it is omitted, 'porter build' fails while trying to parse the bundle reference.

## To Reproduce
1. Run `porter create`
2. Run `porter build`
3. See error

## Expected behavior
I would expect that it would build the bundle using a calculated reference based on 'registry' and 'name' as described in the docs: https://porter.sh/author-bundles/#bundle-metadata 

## Porter Command and Output
```
$ porter create
creating porter configuration in the current directory

$ porter build
Error: invalid bundle reference %!s(<nil>): invalid reference format
```

## Version
porter v0.30.0 (7a18716b)

**Comment by carolynvs** [comment]
I just ran into that last night too! On it 👍

**Comment by carolynvs** [comment]
A fix is available in [v0.30.1](https://github.com/getporter/porter/releases/tag/v0.30.1)
